core/txpool/legacypool/celo_list_test.go

```diff
@@ -0,0 +1,57 @@
+package legacypool
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Tests that transactions can be added to strict lists and list contents and
+// nonce boundaries are correctly maintained.
+func TestStrictCeloListAdd(t *testing.T) {
```

Comment:
I think we need more tests than this especially around total cost calculations.

Reply:
I agree, I was unsure to add those until the l1cost variables are clear